docs/proposals/discovery.md

```diff
@@ -0,0 +1,3 @@
+# Node Discovery
+
+Goal: Decouple discovery from Wireguard to enable decoupled data path drivers.
```

Comment:
For something that's an idea but without a proposal yet, it seems like github issues is sufficient.

Reply:
@nerdalert The same applies to `policy` and `quic`, as well.